src/components/LayoutHeader/Nav/LinkItems/index.tsx

```diff
@@ -45,158 +55,91 @@ const navLinkItemsData: Array<INavLinkData | INavLinkPopupData> = [
+    href: '/community',
+    popupName: 'communityPopup'
   },
   {
     href: '/support',
     eventType: 'support',
     text: 'Support'
   },
+  {
+    href: 'https://learn.iterative.ai/',
+    eventType: 'learn',
+    text: 'Learn'
+  },
   {
     text: 'Other Tools',
-    popup: 'OtherToolsPopup'
+    popupName: 'otherToolsPopup',
+    Popup: props => <OtherToolsPopup {...props} />
+  },
+  {
+    text: '...',
+    popupName: 'otherPopup',
+    Popup: props => <OtherPopup {...props} />,
+    className: styles.other
   }
 ]
 
 const isPopup = (
   item: INavLinkData | INavLinkPopupData
-): item is INavLinkPopupData => (item as INavLinkPopupData).popup !== undefined
-
-const LinkItems: React.FC = ({}) => {
-  const [isCommunityPopupOpen, setIsCommunityPopupOpen] = useState(false)
-  const [isOtherToolsPopupOpen, setIsOtherToolsPopupOpen] = useState(false)
-  const communityPopupContainerEl = useRef<HTMLLIElement>(null)
-  const otherToolsPopupContainerEl = useRef<HTMLLIElement>(null)
-  let pageCloseEventListener: () => void = () => null
-  let keyupCloseEventListener: () => void = () => null
-
-  const closeAllPopups = (): void => {
-    setIsCommunityPopupOpen(false)
-    setIsOtherToolsPopupOpen(false)
-
-    pageCloseEventListener()
-    keyupCloseEventListener()
-  }
-
-  const handlePageClick = (event: MouseEvent): void => {
-    if (
-      !communityPopupContainerEl.current ||
-      !otherToolsPopupContainerEl.current
-    ) {
-      return
-    }
-    if (
-      !communityPopupContainerEl.current.contains(event.target as Node) &&
-      !otherToolsPopupContainerEl.current.contains(event.target as Node)
-    ) {
-      closeAllPopups()
-    }
-  }
-
-  const handlePageKeyup = (event: KeyboardEvent): void => {
-    if (event.key === 'Escape') {
-      closeAllPopups()
-    }
-  }
-
-  const setupPopupEventListeners = (): void => {
-    document.addEventListener('click', handlePageClick)
-    document.addEventListener('keyup', handlePageKeyup)
-
-    pageCloseEventListener = (): void =>
-      document.removeEventListener('click', handlePageClick)
-    keyupCloseEventListener = (): void =>
-      document.removeEventListener('keyup', handlePageKeyup)
-  }
-
-  const openCommunityPopup = (): void => {
-    setupPopupEventListeners()
-    setIsCommunityPopupOpen(true)
-  }
-
-  const openOtherToolsPopup = (): void => {
-    setupPopupEventListeners()
-    setIsOtherToolsPopupOpen(true)
-  }
-
-  const toggleCommunityPopup = (): void => {
-    setIsOtherToolsPopupOpen(false)
-    if (isCommunityPopupOpen) {
-      closeAllPopups()
-    } else {
-      openCommunityPopup()
-    }
-  }
-
-  const toggleOtherToolsPopup = (): void => {
-    setIsCommunityPopupOpen(false)
-    if (isOtherToolsPopupOpen) {
-      closeAllPopups()
-    } else {
-      openOtherToolsPopup()
-    }
+): item is INavLinkPopupData =>
+  (item as INavLinkPopupData).popupName !== undefined
+
+const LinkItems: React.FC = () => {
+  const communityPopup = usePopup()
```

Comment:
I went ahead and created a `usePopup` hook that handles popup functionality. The popup code in the nav was already overly complex and adding a third popup would have made things even more complicated 😅